### PR TITLE
Feature: Provide plugin version

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Provide plugin versioning to daos obtained from subgraph

Task: [OS-215](https://aragonassociation.atlassian.net/browse/OS-215)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder of the package after the [UPCOMING] title and before the latest version.
- [ ] I have tested my code on the test network.

[OS-215]: https://aragonassociation.atlassian.net/browse/OS-215?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mathewmeconry — APPROVED** [review]

File: modules/client/src/interfaces.ts [hunk 1]
```diff
@@ -384,7 +387,12 @@ export type DaoResourceLink = { name: string; url: string };
 export type InstalledPluginListItem = {
   id: string;
   instanceAddress: string;
-  version: string;
+  installations: PluginInstallation[];
+};
+
+export type PluginInstallation = {
```
Comment (brickpop) [inline modules/client/src/interfaces.ts]: Wouldn't the type be `PluginVersion`? or if we don't need to export it, `PluginEntry`?
And the field be `plugins`?

Comment (brickpop) [reply modules/client/src/interfaces.ts]: I see that subgraph uses these names, but face to the developer I don't think they are super accurate

Comment (josemarinas) [reply modules/client/src/interfaces.ts]: mmm, i'd go with `pluginVersions`

File: modules/client/src/internal/utils.ts [hunk 2]
```diff
@@ -96,24 +95,22 @@ export function toDaoListItem(
...
-          // update when subgraph returns version
-          version: "0.0.1",
-        };
+      ): InstalledPluginListItem[] => {
+        return plugin.installations.map((installation) => {
+          return {
+            instanceAddress: plugin.id,
+            id: SubgraphPluginTypeMap.get(
+              plugin.__typename,
+            ) as string,
```
Comment (brickpop) [inline modules/client/src/internal/utils.ts]: What is this value? Is it the pluginRepo ENS?

**Review by brickpop — APPROVED** [review]

File: modules/client/src/internal/utils.ts (hunk 2 above)
Comment (josemarinas) [reply modules/client/src/internal/utils.ts]: Yes, this gets a plugin typename and returns the ens. Now the subdomain can also be queried from subgraph so maybe we can get it from there insead of getting it from a hardcoded map, wdyt?

**Pull request merged**